core/src/test/java/kafka/server/share/SharePartitionTest.java

```diff
@@ -3629,6 +3635,24 @@ public void testAcquisitionLockTimeoutForBatchesPostStartOffsetMovementToMiddleO
             () -> "Acquisition lock never got released.");
     }
 
+    @Test
```

Comment:
I also suggest one more test that does something like this, changes the group config, and then validates that the updated value was applied for the subsequent acquisition lock timeout.